### ISSUE TITLE
Split lines by '\n' from external file or external variable does not work

### ISSUE THREAD

**Issue by Kopfbremse** [mikefarah/yq#1430]
**Version of yq**: 4.30.2
**Operating system**: linux
**Installed via**: binary release

**Describe the bug**
Split lines from external file or external variable does not work.
This makes the processing of external line-delimited sources impossible in some cases.

**Input file `lines.txt`**
```txt
line 1
line 2
```

**Commands**
```bash
export LINES_FROM_VARIABLE='
line 1
line 2
'
export LINES_FROM_FILE='lines.txt'

echo --- 1#4: split lines ---
yq -n -o json '"line 1\nline 2" | split("\n")'                    # working

echo --- 2#4: split lines from internal variable ---
yq -n -o json '"lines 1\nlines 2" as $var | $var | split("\n")'   # working

echo --- 3#4: split lines from external variable ---
yq -n -o json 'env(LINES_FROM_VARIABLE) | split("\n")'            # fails
yq -n -o json 'strenv(LINES_FROM_VARIABLE) | split("\n")'         # fails

echo --- 4#4: split lines from file ---
yq -n -o json 'load_str(env(LINES_FROM_FILE)) | split("\n")'      # fails
```

**Output**
```txt
--- 1#4: split lines ---
[
  "line 1",
  "line 2"
]
```

```txt
--- 2#4: split lines from internal variable ---
[
  "lines 1",
  "lines 2"
]
```

```txt
--- 3#4: split lines from external variable ---
[
  "line 1 line 2"
]
[
  "\nline 1\nline 2\n"
]
```

```txt
--- 4#4: split lines from file ---
[
  "line 1\nline 2\n"
]
```

**Expected behaviour**
Output should be for each command
```txt
[
  "line 1",
  "line 2"
]
```

**Comment by mikefarah** [comment]
Nice find - will fix :)

**Comment by mikefarah** [comment]
Fixed in 4.30.3 - note that you will need to use `strenv` and not `env` (as env passes the content as-is to the yaml decoder, and that strips some of the newlines)
